.circleci/config.yml

```diff
@@ -97,12 +97,90 @@ jobs:
             coverage run --source=super_gradients -m unittest tests/deci_core_unit_test_suite_runner.py
             coverage report
             coverage html  # open htmlcov/index.html in a browser
-
       - store_artifacts:
           path: htmlcov
 
       - store_artifacts:
           path: ~/sg_logs
+      
+  docker_build:
+    parameters:
+        version:
+          type: string
+          default: latest
+    environment: 
+      IMAGE_TAG: << parameters.version >>
+    description: "building the docker"
+    machine: true 
+    resource_class: deci-ai/research-standard
+    steps:
+      - checkout
+      - run:
+         name: making sure we have docker permmissions #ugly but working solution
+         command: sudo chmod 666 /var/run/docker.sock
+      - deci-common/build_docker_image:
+          image_tag: << parameters.version >>
+          build_args: VERSION=${CIRCLE_BRANCH}
+          dockerfile: "scripts/Dockerfile"
+          repository_name: "deciai/super-gradients"
+      - run:
+          name: Archive Docker image
+          command: docker save -o image.tar deciai/super-gradients:${IMAGE_TAG} 
+      - persist_to_workspace:
+           root: .
+           paths:
+             - ./image.tar
+
+
+  dockerhub_push:
+    description: "pushing the docker"
+    parameters:
+        version:
+          type: string
+    environment: 
+      IMAGE_TAG: << parameters.version >>
+      IMAGE_NAME: deciai/super-gradients
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    resource_class: small
+    steps:
+       - attach_workspace:
+          at: /tmp/workspace
+       - run:
+          name: Load archived Docker image
+          command: docker load -i /tmp/workspace/image.tar
```

Comment:
why not one job that buildsm and push it by parameter?

Reply:
to have it non blocking and i don't always want to push.